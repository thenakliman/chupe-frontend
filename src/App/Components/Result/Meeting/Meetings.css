.meeting-container {
  margin: 2rem 20rem;
}

.meeting-create-container {
  display: flex;
  justify-content: space-between;
  margin: 2rem auto;
}

.meeting-input {
  height: 3rem;
  width: 53rem;
  font-size: 1.5rem;
}

.meeting-button {
  padding: 0 1rem;
}