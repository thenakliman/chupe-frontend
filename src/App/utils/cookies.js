import Cookies from 'universal-cookie';
import {verify, decode} from 'jsonwebtoken';

const TOKEN_COOKIE_KEY = 'token';
const TOKEN_EXPIRY_COOKIE_KEY = 'expiryTime';
const TOKEN_USERNAME_KEY = 'username';
const audience = 'chupe-frontend';
const issuer = 'chupe';
export let cookies = new Cookies();
const Config = require('Config');

const setCookie = (key, value) => {
  cookies.set(key, value);
};

export const setCookies = (token) => {
  verify(token,
         Config.signingKey,
         {
            algorithms: ['HS512'],
            audience: audience,
            issuer: issuer,
            ignoreExpiration: false,
         },
         (e) => console.log(e)
  );

  const decodedToken = decode(token);
  setToken(token);
  setTokenExpiryTime(decodedToken.exp);
  setUsername(decodedToken.username);
};

const setToken = (token) => {
  setCookie(TOKEN_COOKIE_KEY, token);
};

const setTokenExpiryTime = (expiryTime) => {
  setCookie(TOKEN_EXPIRY_COOKIE_KEY, expiryTime);
};

const setUsername = (userName) => {
  setCookie(TOKEN_USERNAME_KEY, userName);
};

export const getUsername = () => {
  return cookies.get(TOKEN_USERNAME_KEY);
};

export const getToken = () => {
  if (!cookies.get(TOKEN_COOKIE_KEY) ||
      !cookies.get(TOKEN_EXPIRY_COOKIE_KEY) ||
      cookies.get(TOKEN_EXPIRY_COOKIE_KEY) < Date.now()) {
    return null;
  }

  return cookies.get(TOKEN_COOKIE_KEY);
};
