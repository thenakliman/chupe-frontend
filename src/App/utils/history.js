import {createHashHistory} from 'history';

export let history = createHashHistory();
