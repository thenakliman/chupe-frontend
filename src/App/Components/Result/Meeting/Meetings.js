import React from 'react';
import propTypes from 'prop-types';
import {history} from '../../../utils/history';

require('./Meetings.css');

export class Meetings extends React.Component {
    constructor(props) {
      super(props);
      this.state = {subject: ''};
      this.createMeeting = this.createMeeting.bind(this);
      this.viewMeeting = this.viewMeeting.bind(this);
    }

    componentDidMount() {
      this.props.getAllMeetings();
    }

    onSubjectChange(newSubject) {
      this.setState({subject: newSubject});
    }

    createMeeting() {
      this.props.createMeeting(this.state.subject);
      this.setState({subject: ''});
    }

    viewMeeting(id) {
      history.push('/meeting/' + id);
    }

    render() {
      return (
      <div id='meeting-container-id'
           className='meeting-container'>
          <div className='meeting-create-container'>
              <input id='meeting-input-id'
                className='meeting-input'
                value={this.state.subject}
                onChange={(event) =>
                            this.onSubjectChange(event.target.value)}
              />
              <button id='meeting-button-id'
                className='meeting-button'
                type='button'
                onClick={this.createMeeting}
              >
                Create Meeting
              </button>
          </div>
          <div>
            <table id='all-meetings-ordered-list'>
              <thead>
                <tr className='meeting-table-header'>
                  <th> Id </th>
                  <th> subject </th>
                  <th> Created By </th>
                </tr>
              </thead>
              <tbody id='all-meeting-table-body-id'
                     className={'meeting-table-body'}>
              {
                this.props.meetings.map((meeting) => (
                  <tr key={`${meeting.id}`}
                      id={`meeting-${meeting.id}`}
                      onClick={() => this.viewMeeting(meeting.id)}>
                    <td> {meeting.id} </td>
                    <td className='meeting-row'>
                            {meeting.subject}
                    </td>
                    <td> {meeting.createdBy}</td>
                  </tr>
                ))
              }
              </tbody>
            </table>
          </div>
      </div>
      );
    }
}

Meetings.propTypes = {
  meetings: propTypes.array,
  getAllMeetings: propTypes.func.isRequired,
  createMeeting: propTypes.func.isRequired,
};
