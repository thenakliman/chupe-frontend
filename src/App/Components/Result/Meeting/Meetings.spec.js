/* eslint-disable */
import React from 'react';
import {Meetings} from './Meetings';
/* eslint-enable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import * as History from '../../../utils/history';


const meetings = [{
  id: 10101,
  subject: 'retro - 1',
}, {
  id: 10102,
  subject: 'retro - 2',
}];

describe('Show meetings component', () => {
  it('should call getAllMeetings on component did mount ', () => {
    const getAllMeetings = jest.fn();
    shallow(
        <Meetings
            meetings={meetings}
            getAllMeetings={getAllMeetings}
        />
    );

    expect(getAllMeetings).toHaveBeenCalledWith();
  });

  it('should have rows for each meetings', () => {
    const getAllMeetings = jest.fn();
    const wrapper = shallow(
        <Meetings
            meetings={meetings}
            getAllMeetings={getAllMeetings}
        />
    );

    expect(wrapper.find('tr').length).toBe(3);
  });

  it('should have input field for meeting create', () => {
    const getAllMeetings = jest.fn();
    const wrapper = shallow(
        <Meetings
            meetings={meetings}
            getAllMeetings={getAllMeetings}
        />
    );

    expect(wrapper.find('input').length).toBe(1);
  });

  it('should have input field for session create', () => {
    const getAllMeetings = jest.fn();
    const wrapper = shallow(
        <Meetings
            meetings={meetings}
            getAllMeetings={getAllMeetings}
        />
    );

    expect(wrapper.find('#meeting-button-id').length).toBe(1);
  });

  describe('should match snapshot', () => {
    const wrapper = shallow(<Meetings
        meetings={meetings}
        getAllMeetings={jest.fn()}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('should have initial state with empty string', () => {
    const wrapper = shallow(<Meetings
        meetings={meetings}
        getAllMeetings={jest.fn()}
    />);

    expect(wrapper.state().subject).toEqual('');
  });

  describe('should change internal state of the component', () => {
    const wrapper = shallow(<Meetings
        meetings={meetings}
        getAllMeetings={jest.fn()}
    />);

    const subject = 'some subject';
    wrapper.find('#meeting-input-id')
        .simulate('change', {target: {value: subject}});

    expect(wrapper.state().subject).toEqual(subject);
  });

  describe('should change internal state of the component on change', () => {
    const createMeeting = jest.fn();
    const wrapper = shallow(<Meetings
        meetings={meetings}
        getAllMeetings={jest.fn()}
        createMeeting={createMeeting}
    />);

    const subject = 'some subject';
    wrapper.find('#meeting-input-id')
        .simulate('change', {target: {value: subject}});

    wrapper.find('#meeting-button-id').simulate('click');

    expect(createMeeting)
        .toHaveBeenCalledWith(subject);
  });

  describe('should push new route to history', () => {
    const createMeeting = jest.fn();
    const wrapper = shallow(<Meetings
        meetings={meetings}
        getAllMeetings={jest.fn()}
        createMeeting={createMeeting}
    />);
    History.history.push = jest.fn();

    wrapper.find('#meeting-10101').simulate('click');
    expect(History.history.push).toHaveBeenCalledWith('/meeting/10101');
  });

  describe('should change state to empty string on create of session', () => {
    const createMeeting = jest.fn();
    const wrapper = shallow(<Meetings
        meetings={meetings}
        getAllMeetings={jest.fn()}
        createMeeting={createMeeting}
    />);

    const subject = 'some subject';
    wrapper.find('#meeting-input-id')
        .simulate('change', {target: {value: subject}});

    wrapper.find('#meeting-button-id').simulate('click');

    expect(wrapper.state().subject).toBe('');
  });
});
