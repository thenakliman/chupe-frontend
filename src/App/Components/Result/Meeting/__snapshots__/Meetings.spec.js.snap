// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="meeting-container"
  id="meeting-container-id"
>
  <div
    className="meeting-create-container"
  >
    <input
      className="meeting-input"
      id="meeting-input-id"
      onChange={[Function]}
      value=""
    />
    <button
      className="meeting-button"
      id="meeting-button-id"
      onClick={[Function]}
      type="button"
    >
      Create Meeting
    </button>
  </div>
  <div>
    <table
      id="all-meetings-ordered-list"
    >
      <thead>
        <tr
          className="meeting-table-header"
        >
          <th>
             Id 
          </th>
          <th>
             Subject 
          </th>
          <th>
             Created By 
          </th>
        </tr>
      </thead>
      <tbody
        id="all-meeting-table-body-id"
      >
        <tr
          className="selectable-row"
          id="meeting-10101"
          key="10101"
          onClick={[Function]}
        >
          <td>
             
            10101
             
          </td>
          <td
            className="meeting-row"
          >
            retro - 1
          </td>
          <td>
             
          </td>
        </tr>
        <tr
          className="selectable-row"
          id="meeting-10102"
          key="10102"
          onClick={[Function]}
        >
          <td>
             
            10102
             
          </td>
          <td
            className="meeting-row"
          >
            retro - 2
          </td>
          <td>
             
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
